Reason: With multiple tabs opened, Ctrl+F Ctrl+W crashes konqueror/khtml in 4.3.2...
Backport requested by David Faure

From 2653aa94401fe4111fbaf1bc0072dcb6d965d173 Mon Sep 17 00:00:00 2001
From: dfaure <dfaure@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Thu, 8 Oct 2009 14:04:22 +0000
Subject: [PATCH] Backport fix for 207173/209876: Hide the whole widget, not just the view and the
 viewport. Otherwise the findbar stays around, receives focus, and this makes the
 partmanager activate the part that we're currently deleting...
 Forgot to do this before 4.3.2, so the fix will be in 4.3.3 :(
 CCBUG: 207173

git-svn-id: svn://anonsvn.kde.org/home/kde/branches/KDE/4.3/kdelibs@1032787 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 khtml/khtml_part.cpp |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/khtml/khtml_part.cpp b/khtml/khtml_part.cpp
index 342d0bb..7ca15b3 100644
--- a/khtml/khtml_part.cpp
+++ b/khtml/khtml_part.cpp
@@ -609,8 +609,7 @@ KHTMLPart::~KHTMLPart()
 
   if ( d->m_view )
   {
-    d->m_view->hide();
-    d->m_view->viewport()->hide();
+    widget()->hide();
     d->m_view->m_part = 0;
   }
 
-- 
1.6.5.rc3.193.gdf7a

